tools/offline_log_viewer/data_directory_inspector.py

```diff
@@ -0,0 +1,885 @@
+                continue
+            entry = CloudStorageCacheEntry(hash=item)
+            cloud_storage_cache.entries.append(entry)
+            inspect_cloud_cache_entry(path, entry, description)
+            continue
+
+        if isfile(path):
+            description.unknown.append(File(path))
+        elif isdir(path):
+            unknown_dir(path, description)
+        else:
+            # path was removed while inspecting
+            pass
+
+
+def get_description(root: str) -> Description:
+    description = Description()
+
+    if isfile(root):
+        description.unknown.append(File(root))
+        return description
+
+    if not isdir(root):
+        # path was removed while inspecting
+        return description
+
+    for item in listdir(root):
+        path = join(root, item)
+        if item == "pid.lock":
+            if not expect_file(path, description):
+                continue
+        elif item == "cloud_storage_cache":
+            if not expect_dir(path, description):
+                continue
+            description.cloud_storage_cache = CloudStorageCache(path=path)
+            inspect_cloud_storage_cache(path, description.cloud_storage_cache,
+                                        description)
+        elif item == "startup_log":
+            if not expect_file(path, description):
+                continue
+            try:
+                description.crash_loop_tracker_file = CrashLoopFile(path)
+                description.crash_loop_tracker_file.size = os.path.getsize(
+                    path)
+            except FileNotFoundError:
+                pass
+        elif item == "config_cache.yaml":
+            if not expect_file(path, description):
+                continue
+            description.config_cache = ConfigCache(path)
+        elif item == ".redpanda_data_dir":
+            if not expect_file(path, description):
+                continue
+            description.data_dir_marker = DataDirMarker(path)
+        elif item == "redpanda":
+            if not expect_dir(path, description):
+                continue
+            inspect_redpanda(path, description)
+        elif item == "kafka":
+            if not expect_dir(path, description):
+                continue
+            inspect_kafka(path, description)
+        elif item == "kafka_internal":
+            if not expect_dir(path, description):
+                continue
+            inspect_kafka_internal(path, description)
+        elif item == "syschecks":
+            if not expect_dir(path, description):
+                continue
+            inspect_syschecks(path, description)
+        else:
+            if isfile(path):
+                description.unknown.append(File(path))
+            elif isdir(path):
+                unknown_dir(path, description)
+            else:
+                # path was removed while inspecting
+                pass
+
+    return description
+
+
+parser = argparse.ArgumentParser(
```

Comment:
if you wrap this into a `def main():` func then this script can simply be imported as a module in other python packages with ducktape rather than pipes.